PRODTRIAGE-3817: Update maintenance_window resource docs (#1095)
mongodb/terraform-provider-mongodbatlas: website/docs/r/maintenance_window.html.markdown

# Resource: mongodbatlas_maintenance_window

> A maintenance window is a pre-determined time frame during which your MongoDB Atlas cluster can be taken offline for maintenance activities, such as patching or upgrades. The `mongodbatlas_maintenance_window` resource allows you to configure the start and end times for this maintenance window, as well as the day(s) of the week during which it should occur.

Reviewer: we should try to avoid passive voice per https://www.mongodb.com/docs/meta/style-guide/writing/use-active-voice/
```suggestion
A maintenance window is a pre-determined time frame during which MongoDB Atlas takes your cluster offline for maintenance activities, such as patching or upgrades. The `mongodbatlas_maintenance_window` resource allows you to configure the start and end times for this maintenance window, as well as the day(s) of the week during which it should occur.
```

Author: super helpful

> By using this resource in Terraform, you can automate the management of your maintenance window settings, ensuring that they are consistent across all of your environments and reducing the risk of human error. This can help ensure that your MongoDB Atlas clusters are available when your applications need them.

Reviewer: an imperative is more direct here (see https://www.mongodb.com/docs/meta/style-guide/writing/write-to-the-user/)

also suggestion a simpler verb form in the last sentence
```suggestion
Use this Terraform resource to automate the management of your maintenance window settings, ensuring that they are consistent across all of your environments and reducing the risk of human error. This ensures that your MongoDB Atlas clusters are available when your applications need them.
```

* `project_id` - The unique identifier of the project for the Maintenance Window.
* `day_of_week` - Day of the week when you would like the maintenance window to start as a 1-based integer: S=1, M=2, T=3, W=4, T=5, F=6, S=7.
* `hour_of_day` - Hour of the day when you would like the maintenance window to start. This parameter uses the 24-hour clock, where midnight is 0, noon is 12 (Time zone is UTC).
* `start_asap` - Boolean flag that indicates whether MongoDB Atlas starts the maintenance window immediately upon receiving this request. To start the maintenance window immediately for your project, MongoDB Atlas must have maintenance scheduled and you must set a maintenance window. This flag resets to false after MongoDB Atlas completes maintenance.
* `number_of_deferrals` - Number of times the current maintenance event for this project has been deferred, you can set a maximum of 2 deferrals.

> * `defer` - Defers the maintenance window for the specified project for one week.

Author: ```suggestion
* `defer` - Boolean flag that defers the maintenance window for the specified project for one week.
```

> * `auto_defer` - Toggles automatic deferral of the maintenance window for the specified project for one week.

Reviewer: question: are these two also booleans?

Author: @jwilliams-mongo confirmed. see source code here to these two parameters for more details: https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/mongodbatlas/resource_mongodbatlas_maintenance_window.go#L72

Author: also adjusted slightly to reflect this. feel free to edit as you see fit

> * `auto_defer` - Toggles automatic deferral of the maintenance window for the specified project for one week.

Author: ```suggestion
* `auto_defer` - Boolean flag that toggles automatic deferral of the maintenance window for the specified project for one week.
```
